src/API/Instrumentation/AutoInstrumentation/Instrumentation.php

```diff
@@ -0,0 +1,17 @@
+<?php
+
+declare(strict_types=1);
+
+namespace OpenTelemetry\API\Instrumentation\AutoInstrumentation;
+
+use OpenTelemetry\Config\SDK\Configuration\Context;
+use OpenTelemetry\Context\ContextStorageInterface;
+
+interface Instrumentation
+{
+    /**
+     * @todo context is nullable in order to support autoloading (and retrieving lazy-loaded tracers), but auto and non-auto
```

Comment:
there's a race condition when autoloading/auto-registering an instrumentation vs loading it later (see the two examples, configure_instrumentation.php and configure_instrumentation_global.php). For the moment, this allows a context or will use Globals - we should choose one approach.

Reply:
I've gone with using globals, as we currently do. Autoloading happens too early for providers to be available to pass in via context.